File: src/ru/luk/utils/PlayerItemInHandItemDamage.java
package ru.luk.utils;

import org.bukkit.Sound;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Player;

import java.util.Arrays;
import java.util.List;


public class PlayerItemInHandItemDamage {
    private static final List<List<Integer>> chanceList = Arrays.asList(
            Arrays.asList(0, 0, 0, 1),
            Arrays.asList(0, 0, 1, 1),
            Arrays.asList(0, 1, 1, 1)
    );

    public static void playerItemInHandItemDamage(Player player) {
        if (player.getInventory().getItemInMainHand().containsEnchantment(Enchantment.DURABILITY)) {
            int level = player.getInventory().getItemInMainHand().getEnchantmentLevel(Enchantment.DURABILITY);

            if (chanceList.get(level - 1).get((int) (Math.random() * 3)) == 1) {
                return;
            }
        }

        org.bukkit.inventory.meta.Damageable damageableMeta = (org.bukkit.inventory.meta.Damageable) player.getInventory().getItemInMainHand().getItemMeta();
        int damage = damageableMeta.getDamage();
        int maxDurability = player.getInventory().getItemInMainHand().getType().getMaxDurability();
        damageableMeta.setDamage(damage + 1);

        if ((damage + 1) < maxDurability) {
            player.getInventory().getItemInMainHand().setItemMeta(damageableMeta);

        } else {
            player.playSound(player.getLocation().add(0, -1, 0), Sound.ENTITY_ITEM_BREAK, 1, 1);
            player.getInventory().getItemInMainHand().setAmount(0);
        }

        player.updateInventory();
    }
}

File: src/main/kotlin/ru/luk/utils/PlayerItemInHandItemDamage.kt
package ru.luk.utils

import org.bukkit.Sound
import org.bukkit.entity.Player
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.meta.Damageable
import org.bukkit.inventory.meta.ItemMeta

import kotlin.random.Random


private val levelChancesMap: Map<Int, Int> = mapOf(
    1 to 25,
    2 to 36,
    3 to 43
)

fun playerItemInHandItemDamage(player: Player) {
    if (Enchantment.DURABILITY in player.inventory.itemInMainHand.enchantments) {
        if (Random.nextInt(
                0,
                100
            ) in 0..levelChancesMap[player.inventory.itemInMainHand.getEnchantmentLevel(Enchantment.DURABILITY)]!!
        ) {
            return
        }
    }

    val damageableMeta: Damageable = player.inventory.itemInMainHand.itemMeta as Damageable
    damageableMeta.damage += 1
    val maxDurability: Short = player.inventory.itemInMainHand.type.maxDurability

    if (damageableMeta.damage < maxDurability) {
        player.inventory.itemInMainHand.itemMeta = damageableMeta as ItemMeta
    } else {
        player.playSound(player.location.add(.0, -1.0, .0), Sound.ENTITY_ITEM_BREAK, 1F, 1F)
        player.inventory.itemInMainHand.amount = 0
    }
    player.updateInventory()
}